src/frontend/js/widgets/Dashboard/components/DashboardItem/Order/ContractFrame/index.tsx

```diff
@@ -0,0 +1,253 @@
+      shouldCloseOnEsc={false}
+      onRequestClose={props.onClose}
+      testId={'contract-frame__' + props.order.id}
+    >
+      {/* We want to have a fresh component when we close and re-open the modal to avoid any */}
+      {/* state resetting issues. */}
+      {props.isOpen && <ContractFrameContent {...props} />}
+    </Modal>
+  );
+};
+
+export const ContractFrameContent = ({ order, onClose, onDone, product }: Props) => {
+  const api = useJoanieApi();
+  const intl = useIntl();
+  const [step, setStep] = useState(ContractSteps.LOADING_CONTRACT);
+  const [signatureType, setSignatureType] = useState<SignatureType>();
+  const [invitationLink, setInvitationLink] = useState<Maybe<string>>();
+  const [error, setError] = useState<Maybe<string>>();
+  const queryClient = useQueryClient();
+  const timeoutRef = useRef<NodeJS.Timeout>();
+
+  const setErrored = (e: string) => {
+    setStep(ContractSteps.ERROR);
+    setError(e);
+  };
+
+  const initialize = async () => {
+    try {
+      const response = await api.user.orders.submit_for_signature(order.id);
+      await stepSign(response.invitation_link);
+    } catch (e) {
+      setErrored(intl.formatMessage(messages.errorSubmitForSignature));
+    }
+  };
+
+  const stepSign = async (invitationLink_: string) => {
+    setInvitationLink(invitationLink_);
+    if (invitationLink_.match(DUMMY_REGEX)) {
+      // Nothing to do, in dummy mode submitting for signature automatically signs the contract.
+      setSignatureType(SignatureType.DUMMY);
+    } else {
+      // Open iframe then wait for it to close.
+      setSignatureType(SignatureType.LEX);
+    }
+    setStep(ContractSteps.SIGNING);
+  };
+
+  const onSigned = () => {
+    stepPoll();
+  };
+
+  // Polling the order from the backend waiting for it to have processed the inbound lex persona
+  // webhook.
+  const stepPoll = async () => {
+    setStep(ContractSteps.POLLING);
+    let round = 0;
+    const checkOrderSignature = async () => {
+      if (round >= CONTRACT_SETTINGS.pollLimit) {
+        timeoutRef.current = undefined;
+        setErrored(intl.formatMessage(messages.errorMaxPolling));
+      } else {
+        try {
+          const orderToCheck = (await api.user.orders.get({ id: order.id })) as Order;
+          const isSigned = !orderNeedsSignature(orderToCheck, product);
+          if (isSigned) {
+            timeoutRef.current = undefined;
+            stepFinished();
+          } else {
+            round++;
+            timeoutRef.current = setTimeout(checkOrderSignature, CONTRACT_SETTINGS.pollInterval);
+          }
+        } catch (e) {
+          setErrored(intl.formatMessage(messages.errorPolling));
+        }
+      }
+    };
+
+    checkOrderSignature();
+  };
+
+  const stepFinished = () => {
+    setStep(ContractSteps.FINISHED);
+    queryClient.invalidateQueries({ queryKey: ['user', 'orders'] });
+    onDone();
+  };
+
+  useEffect(() => {
+    initialize();
+  }, []);
```

Comment:
By convention, we put `useEffect` after variable declaration, just before the render statement